src/routers/router.js

```diff
@@ -29,6 +29,7 @@ router.get('/examples', validateApiKey, attachRedisClient, getExamples);
 router.get('/examples/:id', validateApiKey, validId, attachRedisClient, getExample);
 
 router.post('/developers', createDeveloperLimiter, validateDeveloperBody, postDeveloper);
+router.get('/developers/:id', validId, attachRedisClient, getDeveloper);
```

Comment:
make sure to test your changes locally. the middleware `validId` wouldn't work since it only works for MongoDB document Ids. You will probably want to create a new middleware like `authorization` that checks the user's API token (similar to what `validateApiKey` does) but then checks that the provided API key is associated with the requested use with `:id`